segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/UtilsWrapper.java

```diff
@@ -0,0 +1,287 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.segmentstore.storage.chunklayer;
+
+import com.google.common.base.Preconditions;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.segmentstore.storage.metadata.BaseMetadataStore;
+
+import io.pravega.segmentstore.storage.metadata.SegmentMetadata;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+
+/**
+ * This class contains various utils methods useful for administration of LTS.
+ */
+@Slf4j
+@AllArgsConstructor
+@Data
+public class UtilsWrapper {
+    private static final SystemJournal.SystemJournalRecordBatch.SystemJournalRecordBatchSerializer BATCH_SERIALIZER = new SystemJournal.SystemJournalRecordBatch.SystemJournalRecordBatchSerializer();
+
+    @NonNull
+    private ChunkedSegmentStorage chunkedSegmentStorage;
+
+    private int bufferSize;
+
+    @NonNull
+    private Duration timeout;
+
+    /**
+     * Evict metadata cache.
+     *
+     * @return A CompletableFuture that, when completed, will indicate that the operation completed.
+     *         If the operation failed, it will be completed with the appropriate exception.
+     */
+    public CompletableFuture<Void> evictMetadataCache() {
```

Comment:
The scenario is when metadata is changed in the storage but old values are still cached.
Instead of evicting individual values we just nuke the whole cache.

Reply:
updated comment.